src/map.cpp

```diff
@@ -5595,10 +5595,9 @@ void map::update_visibility_cache( visibility_variables &cache, const int zlev )
     cache.u_clairvoyance = g->u.clairvoyance();
     cache.u_sight_impaired = g->u.sight_impaired();
     cache.u_is_boomered = g->u.has_effect("boomered");
-
-    int sm_squares_seen[my_MAPSIZE][my_MAPSIZE];
-    std::memset(sm_squares_seen, 0, sizeof(sm_squares_seen));
-
+   
+    std::vector<std::vector<int>> sm_squares_seen(my_MAPSIZE, std::vector<int>(my_MAPSIZE));
```

Comment:
In all of these cases we can just switch to using the MAPSIZE constant instead of my_MAPSIZE.